adapter-base/src/main/java/org/eclipse/hono/adapter/auth/device/jwt/CredentialsParser.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.adapter.auth.device.jwt;
+
+import java.net.HttpURLConnection;
+import java.util.Optional;
+
+import org.eclipse.hono.client.ClientErrorException;
+import org.eclipse.hono.util.CredentialsConstants;
+
+import io.jsonwebtoken.Claims;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * A service for extracting and parsing tenant and device information.
+ *
+ */
+public interface CredentialsParser {
+
+    /**
+     * Extracts the tenantId and authId from JWS Claims.
+     *
+     * @param claims A JsonObject containing the JWS Claims.
+     * @return A JsonObject containing the tenantId and authId extracted from the JWS Claims.
+     * @throws ClientErrorException If tenantId or authId cannot correctly be extracted from the JWS Claims.
```

Comment:
how about adding an explicit `null` check for the passed in claims?

Reply:
I am not quite sure I understand what you mean exactly. Do you mean adding a check if the input JsonObject is null or if the "tid" and "sub" claims are null? The latter we have already.

Reply:
We usually document if a method throws a NPE if one of the input parameters is `null` (the *claims* param in this case). IMHO you should also add an explicit check and throws expression instead of relying on the implicit NPE when accessing one of claims' fields/methods.